02-tamagotchi-interaction/src/lib.rs

```diff
@@ -28,17 +48,72 @@ extern "C" fn handle() {
     };
     match action {
         TmgAction::Name => {
-            debug!("TmgAction::Name {:?}", tamagotchi.name);
             msg::reply(TmgEvent::Name(tamagotchi.name.clone()), 0)
                 .expect("Error in sending reply TmgEvent::Name");
         }
         TmgAction::Age => {
             let age = exec::block_timestamp() - tamagotchi.date_of_birth;
-            debug!("TmgAction::Age {:?}", age);
             msg::reply(TmgEvent::Age(age), 0).expect("Error in sending reply TmgEvent::Age");
         }
+        TmgAction::Feed => {
+            // To determine the number of blocks when the Tamagotchi last ate
+            let current_block: u64 = exec::block_height().into();
+
+            // we get to update the fed and fed_block
+
+            // we use saturating_sub so that the subtraction result won't be below 0
+            let fed = tamagotchi
+                .fed
+                .saturating_sub((current_block - tamagotchi.fed_block) * HUNGER_PER_BLOCK);
+
+            // calculate amount needed to feed
+            let fed = fed + FILL_PER_FEED;
+
+            // we make sure that amount must be in correct range
+            let fed = cmp::min(fed, MAX_MOOD_VALUE);
+            let fed = cmp::max(fed, MIN_MOOD_VALUE);
```

Comment:
Using [`u64::clamp`](https://doc.rust-lang.org/std/cmp/trait.Ord.html#method.clamp) makes code shorter:
```suggestion
            let fed = fed.clamp(MIN_MOOD_VALUE, MAX_MOOD_VALUE);
```